spanner/transaction.js

```diff
@@ -96,53 +87,47 @@ function readWriteTransaction (instanceId, databaseId) {
   const instance = spanner.instance(instanceId);
   const database = instance.database(databaseId);
 
-  // Gets a transaction object that captures the database state
-  // at a specific point in time
-  let transaction, firstBudget, secondBudget;
   const transferAmount = 200000;
   const minimumAmountToTransfer = 300000;
 
-  database.runTransaction()
-    .then((results) => {
-      transaction = results[0];
-
-      const queryOne = {
-        columns: [`MarketingBudget`],
-        keys: [2, 2] // SingerId: 2, AlbumId: 2
-      };
-
-      const queryTwo = {
-        columns: ['MarketingBudget'],
-        keys: [1, 1] // SingerId: 1, AlbumId: 1
-      };
-
-      return Promise.all([
-        // Reads the second album's budget
-        transaction.read('Albums', queryOne).then((results) => {
-          // Gets second album's budget
-          // Note: MarketingBudget is an INT64, which comes from Cloud Spanner
-          // as a string - so we convert it to a number with parseInt()
-          const rows = results[0].map((row) => row.toJSON());
-          secondBudget = parseInt(rows[0].MarketingBudget.value);
-          console.log(`The second album's marketing budget: ${secondBudget}`);
-
-          // Makes sure the second album's budget is sufficient
-          if (secondBudget < minimumAmountToTransfer) {
-            throw new Error(`The second album's budget (${secondBudget}) is less than the minimum required amount to transfer.`);
-          }
-        }),
-
-        // Reads the first album's budget
-        transaction.read('Albums', queryTwo).then((results) => {
-          // Gets first album's budget
-          // As above, MarketingBudget is an INT64 and comes as a string
-          const rows = results[0].map((row) => row.toJSON());
-          firstBudget = parseInt(rows[0].MarketingBudget.value);
-          console.log(`The first album's marketing budget: ${firstBudget}`);
-        })
-      ]);
-    })
-    .then(() => {
+  database.runTransaction(function(err, transaction) {
```

Comment:
database.runTransaction no longer returns a Promise. This was a change made in the latest release